website/docs/plugin/framework/deprecations.mdx

```diff
@@ -0,0 +1,1282 @@
+
+The procedure for renaming an attribute depends on what type of attribute it is:
+
+- [Renaming a Required Attribute](#renaming-a-required-attribute)
+- [Renaming an Optional Attribute](#renaming-an-optional-attribute)
+- [Renaming a Computed Attribute](#renaming-a-computed-attribute)
+
+### Renaming a Required Attribute
+
+~> **NOTE:** If the schema definition does not contain `Optional` or `Required`, see the [Renaming a Computed Attribute section](#renaming-a-computed-attribute) instead. If the schema definition contains `Optional` instead of `Required`, see the [Renaming an Optional Attribute section](#renaming-an-optional-attribute).
+
+-> [Required attributes](/terraform/plugin/framework/handling-data/schemas#required) are also referred to as required "arguments" throughout the Terraform documentation.
+
+In general, the procedure here does two things:
+
+- Prevents the operator from needing to define two attributes with the same value.
+- Allows the operator to migrate the configuration to the new attribute at the same time requiring that any other references only work with the new attribute. This is to prevent a situation with Terraform showing a difference when the existing attribute is configured, but the new attribute is saved into the Terraform state. For example, in `terraform plan` output format:
+
+```
+existing_attribute: "" => "value"
+new_attribute:      "value" => ""
+```
+
+The recommended process is as follows:
+
+1. Replace `Required: true` with `Optional: true` in the existing attribute schema definition.
+1. Replace `Required` with `Optional` in the existing attribute documentation.
+1. Duplicate the schema definition of the existing attribute, renaming one of them with the new attribute name.
+1. Duplicate the documentation of the existing attribute, renaming one of them with the new attribute name.
+1. Add a [`DeprecationMessage`](/terraform/plugin/framework/handling-data/schemas#deprecationmessage-1) to the schema definition of the existing (now the "old") attribute, noting to use the new attribute in the message.
+1. Add `**Deprecated**` to the documentation of the existing (now the "old") attribute, noting to use the new attribute.
+1. Add a note to the documentation that either the existing (now the "old") attribute or new attribute must be configured.
+1. Add the type-specific [validator](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators) `{type}validator.ExactlyOneOf` to the schema definition of the new attribute, with a path expression matching the old attribute. This will ensure at least one of the attributes is configured, but present an error to the operator if both are configured at the same time. For example, an attribute of type string would use the [`stringvalidator.ExactlyOneOf`](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator#ExactlyOneOf) validator.
```

Comment:
I initially had this as the `ConflictsWith` validator, not sure which one is recommended

Reply:
I believe `ConflictsWith` or `ExactlyOneOf` can be used with equivalent results but I think I would lean towards the latter as seems from a semantic point of view to be more aligned with the intent.